src/state/include/com/amazonaws/kinesis/video/state/Include.h

```diff
@@ -64,6 +64,16 @@ typedef STATUS (*GetNextStateFunc)(UINT64, PUINT64);
  */
 typedef STATUS (*ExecuteStateFunc)(UINT64, UINT64);
 
+/**
+ * State's error handler function. Called before state machine
+ * transitions to a different state.
+ *
+ * @param 1 - IN - Custom data passed in
+ *
+ * @return Status of the callback
+ */
+typedef STATUS (*ErrorHandlerStateFunc)(UINT64);

```

Comment:
This seem to be not an error handler. It's more of the state transition function. We have a very similar concept in the Signaling client in WebRTC.

Recommend changing the name of StateMachineTransitionFunc or something similar.

Reply:
Another question is whether this belongs to the public header. So far, it seems to be used in the internal functionality. Is the idea to expose this API to C producer/WebRTC?

Reply:
I agree that this is technically a state transition function, but we intend to use this for executing some logic for state transitions on errors. StateMachineTransitionFunc sounds more like we need to perform "some" logic while transitioning states. To me adding anything at this layer while transitioning state is a bit risky and we just intend to do error check and add wait time to avoid high level tight loops. So I suggest to stick with ErrorHandlerStateFunc name for it to be very specific.
Thoughts? 

We do not intend to expose this in public APIs. Exposing this to the customers might not be super useful. Customers should be abstracted from state machine implementation. Also exposing such low level APIs might lead customers doesn't seem right (unless there's a strong reason). This is not a one way door though. We can always expose these APIs later if need arise.

Reply:
Seems that this is indeed a hook function. Take a look at the other hooks - those are called BlahBlahHookFn or something similar. I really recommend not using the "Error" and "Handler" prefix here at all. Make sure you have a "Hook" prefix as this will be exposed in the public header. Also, I commented below on whether to take a single API instead of per state callback which might not make sense

Reply:
Sure I'll change it to `StateTransitionHookFunc`. Replying to the other comment below.